pkg/collector/upgrade/v0_38_0.go

```diff
@@ -0,0 +1,46 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package upgrade
+
+import (
+	"fmt"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/open-telemetry/opentelemetry-operator/api/collector/v1alpha1"
+)
+
+func upgrade0_38_0(cl client.Client, otelcol *v1alpha1.OpenTelemetryCollector) (*v1alpha1.OpenTelemetryCollector, error) {
+	// return if args exist
+	if len(otelcol.Spec.Args) == 0 {
+		return otelcol, nil
+	}
+
+	// Remove otelcol args --log-level, --log-profile, --log-format
+	// are deprecated in reference to https://github.com/open-telemetry/opentelemetry-collector/pull/4213
+	var foundLoggingArgs []string
+	for argKey := range otelcol.Spec.Args {
+		if argKey == "--log-level" || argKey == "--log-profile" || argKey == "--log-format" {
```

Comment:
Can/should we translate that into config changes, to accomplish the same? I believe most, or all of those properties are now available in the collector config itself.

Reply:
config changes? I do not see an option in otel collector configuration which includes args for otel collector container. :) @jpkrohling

Reply:
No, I mean that the args from the collector became options in the collector config: https://github.com/open-telemetry/opentelemetry-collector/pull/4009 . Like this:

```
service:
  telemetry:
    logs:
      level: "DEBUG"
      development: true
      encoding: "json"
```